projects/step-core/src/lib/services/function-actions-impl.service.ts

```diff
@@ -138,8 +138,8 @@ export class FunctionActionsImplService implements FunctionActionsService {
   selectFunction(): Observable<Keyword> {
     const selectedEntity$ = this._entityDialogs.selectEntityOfType('function', true);
     const function$ = selectedEntity$.pipe(
-      map((result) => result.item),
-      switchMap((id) => this._functionApiService.getFunctionById(id))
+      map((result) => result.item as Keyword),
+      switchMap((f: Keyword) => this._functionApiService.getFunctionById(f.id!))
```

Comment:
More readable. Typescript will resolve the type automatically, because there is cast at the previous line
```suggestion
      switchMap((keyword) => this._functionApiService.getFunctionById(keyword.id!))
```